recipes/_start.rb

```diff
@@ -10,6 +10,16 @@
   not_if { platform_family?('windows') }
 end
 
+ruby_block 'insert ulimit' do
+  block do
+    file = Chef::Util::FileEdit.new('/etc/init.d/splunk')
+    file.insert_line_after_match(/^RETVAL=\d$/, "ulimit -n #{node['splunk']['limits']['open_files']}")
+    file.write_file
+  end
+  not_if { platform_family?('windows') || File.readlines('/etc/init.d/splunk').grep(/ulimit/).any? }
```

Comment:
So you're correct in that the boot-start command re-writes the file every time, so this would edit the file every time.

But that turns line 20 into a problem, as we'd only want to mark the service for a restart if the ulimit has changed. (we'd wind up restarting every time here)... Possibly have to look at the file and set the marker beforehand.